### PR TITLE
docs: update rollout and data targeting

### PR DESCRIPTION
**Pull request by heitortessaro**

### CODE REVIEW

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+slug: /feature-flags/creating-feature-flags/auto-operation/rollout
+description: 'The rollout features'
+tags: ['guide', 'automation', 'rollout', 'feature-flag']
+---
+
+import CenteredImg from '@site/src/components/centered-img/CenteredImg';
+
+The rollout solution is designed to progressively deliver new features for your users. It automates the release process, gradually providing the flag variation you choose to a larger percentage of your users to ensure it performs as expected. You can use this approach to detect problems earlier and reduce the impact of any issue.
+
+When you choose to use the rollout option, you are selecting an enabling operation from the [Auto Operation](/feature-flags/creating-feature-flags/auto-operation). That means that you will choose the flag variation that more users will receive as time goes on and the rollout approaches its end.
```
Comment (cre8ivejp) [inline docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:13]:
```suggestion
When you use the rollout option, you select an enabling operation from the [Auto Operation](/feature-flags/creating-feature-flags/auto-operation). That means that you will choose the flag variation that more users will receive as time goes on and the rollout approaches its end.
```

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+- **Variation**: Choose the variation to roll out. You should not choose the same variation selected as [off variation](/feature-flags/creating-feature-flags/targeting#the-targeting-page). Otherwise, you will receive an error message.
+- **Weight**: The percentual value of users receiving the selected variation at the selected date. 
+- **Execute at**: The date and time when the rollout phase will be released, updating the percentage of users accessing the selected variation.
+
+Using the Manual rollout, you do not need to follow an incremental rule for each rollout phase. The only restriction is that each new rollout phase must have a higher weight than the previous one. 
+
+Another important point related to the Manual option is that you don't need to reach 100% to submit the rollout. Supose the last entry to the Manual configuration is 90%. In this case, the rollout will be concluded, and only 90% of your users will have access to the new variation. The remaining 10% will continue receiving the [off variation](/feature-flags/creating-feature-flags/targeting#the-targeting-page), the same as before the rollout.
```
Comment (cre8ivejp) [inline docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:82]:
```suggestion
Another important point related to the Manual option is that you don't need to reach 100% to submit the rollout. Suppose the last entry to the Manual configuration is 90%. In this case, the rollout will be concluded, and only 90% of your users will have access to the new variation. The remaining 10% will continue receiving the [off variation](/feature-flags/creating-feature-flags/targeting#the-targeting-page), the same as before the rollout.
```

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+title: Progressive rollout
+# sidebar_position:
+slug: /feature-flags/creating-feature-flags/auto-operation/progressive-rollout
+description: 'Check how the progressive rollout features work in Bucketeer'
+tags: ['guide', 'automation', 'rollout', 'progressive rollout']
+---
+
+import CenteredImg from '@site/src/components/centered-img/CenteredImg';
+
+The Progressive Rollout solution automates the release process of new features to your users by gradually providing the flag variation you choose as the [default strategy](/feature-flags/creating-feature-flags/targeting) to a larger percentage of your users. This approach helps you detect problems early on and reduce the impact.
```
Comment (cre8ivejp) [inline docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:11]:
```suggestion
The Progressive Rollout solution automates the release process of new features to your users by gradually providing the flag variation you choose as the [default strategy](/feature-flags/creating-feature-flags/targeting) to a larger percentage of your users. This approach helps you detect problems early on and reduce the impact.
```

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+tags: ['guide', 'automation', 'rollout', 'progressive rollout']
+---
+
+import CenteredImg from '@site/src/components/centered-img/CenteredImg';
+
+The Progressive Rollout solution automates the release process of new features to your users by gradually providing the flag variation you choose as the [default strategy](/feature-flags/creating-feature-flags/targeting) to a larger percentage of your users. This approach helps you detect problems early on and reduce the impact. 
+
+## How rollout works
+
+You need to set two variation flags to use the rollout feature. For example, consider that you have a boolean flag with two variations: `true` and `false`. This flag determines whether new content is available to users or not. Currently, the `false` variation is the [OFF variation](/feature-flags/creating-feature-flags/targeting#the-targeting-page), meaning users can't see the new content. However, you want to make the content available to all users within ten days. In this case, you can use the rollout feature to progressively roll out the `true` variation to 10% of your users daily.
```
Comment (cre8ivejp) [inline docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:15]:
```suggestion
The flag must have **two variations** to use the rollout feature. For example, consider that you have a boolean flag with two variations: `true` and `false`. This flag determines whether new content is available to users or not. 

Currently, the `false` variation is the [OFF variation](/feature-flags/creating-feature-flags/targeting#the-targeting-page), meaning users can't see the new content. However, you want to make the content available to all users within ten days. In this case, you can use the rollout feature to progressively roll out the `true` variation to 10% of your users daily. 
```

**Review by cre8ivejp — CHANGES_REQUESTED** [review]

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx (hunk 3 above)
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:11]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx (hunk 4 above)
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:15]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+## Creating a rollout
+
+To add a rollout to a feature flag, you need to follow the steps below:
+
+1. Go to the **Auto Operation** page.
+2. Click the **+ Add** button.
+3. The operation setting panel will appear. Choose the **Enable** operation.
+4. Select **Progreesive Rollout**.
+5. Set up the rollout.
+6. Click **Submit**.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:35]: @cre8ivejp, Should I delete steps from 3 to 6?

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+:::caution Number of variations
+
+The progressive rollout only works for flags with two variations. 
+
+:::
+
+:::caution Disabled Flag behaviour
+
+If a progressive rollout is created for a disabled flag, the progressive rollout automatically enables the flags as soon as it starts. The progressive rollout can only enable a flag once, when it starts.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:45]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+If a flag with an active progressive rollout is turned OFF, the system will automatically stop the progressive rollout.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:47]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+### Restriction when updating a flag
+
+If you need to update a flag that has a running progressive rollout associated with it, be aware of the following restrictions:
+
+- The default strategy on the [Targeting](/feature-flags/creating-feature-flags/targeting) tab can't be modified.
+- You're not able to add or remove variations to the flag. However, you can edit the values, names, and descriptions of the existing variations.
+
+:::info Updating a rollout
+
+Editing a running progressive rollout is not supported in Bucketeer. You can remove or stop the rollout.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:125]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+
+If you remove the rollout, no historical data will be stored. Only logs will report the operations related to that rollout.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:127]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+
+When you stop the rollout, it's shut, moving it to the finished tab, where you can inspect the rollout level it reached, for example.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:129]: Done

[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:129]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+
+:::
+
+### Operational rules
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:133]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+Progressive rollouts can be created for flags, whether they're enabled or disabled. If a progressive rollout is created for a disabled flag, the progressive rollout will automatically enable the flags as soon as it starts.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:135]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+
+It's vital to notice that executing a Kill Switch, triggered by schedule or event rate conditions, has higher priority than the progressive rollout. Therefore, if your flag has a running progressive rollout and a Kill Switch operation is triggered, the system will stop the progressive rollout. The progressive rollout will be moved to the **Finished** tab, where you can check the rollout evolution before being stopped.
+
+## Restrictions and rules
+
+Creating and using progressive rollout features come with certain restrictions to ensure smooth implementation and usage.
+
+### Restrictions when creating a rollout
+
+The progressive rollout is designed to work exclusively with flags having two variations. If you try to use it with a flag with more variations, you'll receive an error message during the progressive rollout creation process.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:106]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+
+If you create a progressive rollout and the flag default strategy uses a [rollout percentage](/feature-flags/creating-feature-flags/targeting#rollout-percentage), the existing rollout rules will be removed. The default strategy will be reset according to the settings defined during the progressive rollout creation.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:108]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+  alt="Template rollout"
+  wSize="400px"
+  borderWidth="1px"
+/>
+
+## Combining auto operations
+
+You can combine the usage of available [auto operations](/feature-flags/creating-feature-flags/auto-operation) from Bucketeer. For example, you can define a progressive rollout, a [schedule](/feature-flags/creating-feature-flags/auto-operation/schedule), and a [event rate](/feature-flags/creating-feature-flags/auto-operation/event-rate) conditions for the same flag. 
+
+It's vital to notice that executing a Kill Switch, triggered by schedule or event rate conditions, has higher priority than the progressive rollout. Therefore, if your flag has a running progressive rollout and a Kill Switch operation is triggered, the system will stop the progressive rollout. The progressive rollout will be moved to the **Finished** tab, where you can check the rollout evolution before being stopped.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:98]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+  alt="Template rollout"
+  wSize="400px"
+  borderWidth="1px"
+/>
+
+### Manual rollout
+
+When using the Manual rollout, the configurations required to submit the rollout differ from those used on the Template rollout. In this case, you need to set the following parameters:
+
+- **Variation**: Choose the variation to roll out. You should not choose the same variation selected as [OFF variation](/feature-flags/creating-feature-flags/targeting#the-targeting-page). Otherwise, you will receive an error message.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:77]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+The rollout provides two main options when setting it up:
+
+- [**Template**](/feature-flags/creating-feature-flags/auto-operation/progressive-rollout#template-rollout): With this option, you define the percentual increment and the frequency of the rollout. The weight of each phase and the execution time are automatically defined until the rollout reaches 100%.
+- [**Manual**](/feature-flags/creating-feature-flags/auto-operation/progressive-rollout#manual-rollout): In this case, you are responsible for defining each rollout phase's percentual weight and release time.
+
+### Template rollout
+
+If you choose to use the Template rollout, you need to define the following configurations before submitting it:
+
+- **Variation**: Choose the variation to roll out. You should not choose the same variation selected as [OFF variation](/feature-flags/creating-feature-flags/targeting#the-targeting-page). Otherwise, you will receive an error message.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:59]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+1. Go to the **Auto Operation** page.
+2. Click the **+ Add** button.
+3. The operation setting panel will appear. Choose the **Enable** operation.
+4. Select **Progreesive Rollout**.
+5. Set up the rollout.
+6. Click **Submit**. 
+
+:::caution Number of variations
+
+The progressive rollout only works for flags with two variations.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:39]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+  alt="Rollout example"
+  wSize="550px"
+  borderWidth="1px"
+/>
+
+## Creating a rollout
+
+To add a rollout to a feature flag, you need to follow the steps below:
+
+1. Go to the **Auto Operation** page.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:30]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 20]
```diff
@@ -0,0 +1,137 @@
...
+2. Click the **+ Add** button.
+3. The operation setting panel will appear. Choose the **Enable** operation.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:32]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+4. Select **Progreesive Rollout**.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:33]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:35]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 22]
```diff
@@ -0,0 +1,137 @@
...
+
+### Manual rollout
+
+When using the Manual rollout, the configurations required to submit the rollout differ from those used on the Template rollout. In this case, you need to set the following parameters:
+
+- **Variation**: Choose the variation to roll out. You should not choose the same variation selected as [OFF variation](/feature-flags/creating-feature-flags/targeting#the-targeting-page). Otherwise, you will receive an error message.
+- **Weight**: The percentual value of users receiving the selected variation at the selected date. 
+- **Execute at**: The date and time when the rollout phase will be released, updating the percentage of users accessing the selected variation.
+
+Using the Manual rollout, you do not need to follow an incremental rule for each rollout phase. The only restriction is that each new rollout phase must have a higher weight than the previous one.
```
[the remark this answers is not in the document]
Comment (heitortessaro) [reply docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:81]: Done

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 23]
```diff
@@ -0,0 +1,141 @@
...
+
+:::caution Number of variations
+
+The progressive rollout only works for flags with **two variations**. 
+
+:::
+
+:::caution Disabled Flag behaviour
+
+If the flag is disabled when a progressive rollout is created, the progressive rollout automatically enables the flag as soon as it starts. The progressive rollout can only enable a flag once when it starts.
```
Comment (cre8ivejp) [inline docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:45]:
```suggestion
If the flag is disabled when a progressive rollout is created, the progressive rollout automatically enables the flag as soon as it starts.
```

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 24]
```diff
@@ -0,0 +1,141 @@
...
+
+It's vital to note that turning off a flag by schedule or event rate conditions has a higher priority than the progressive rollout. Therefore, if your flag has a running progressive rollout and an Auto Operation is triggered, the system will stop the progressive rollout. The progressive rollout will be moved to the **Finished** tab, where you can check the last rollout state or use it as a reference for future rollouts.
+
+## Restrictions and rules
+
+Creating and using progressive rollout features come with certain restrictions to ensure smooth implementation and usage.
+
+### Restrictions when creating a rollout
+
+The progressive rollout is designed to work exclusively with flags having two variations.
```
Comment (cre8ivejp) [inline docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:110]:
```suggestion
The progressive rollout is designed to work exclusively with flags having two variations and no running experiment.
```

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 25]
```diff
@@ -0,0 +1,141 @@
...
+## Creating a rollout
+
+To add a rollout to a feature flag, you need to follow the steps below:
+
+1. Go to the **Auto Operation** tab on the feature Flag page details.
+2. Click the **+ Add** button.
+3. Choose the **Enable** operation and select **Progressive Rollout**.
+4. Configure the rollout and submit.
+
+:::caution Number of variations
```
Comment (cre8ivejp) [inline docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:37]:
```suggestion
:::caution Restrictions
```

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 26]
```diff
@@ -0,0 +1,141 @@
...
+
+The progressive rollout only works for flags with **two variations**.
```
Comment (cre8ivejp) [inline docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:39]: The progressive rollout only works for flags with **two variations** and **no running experiment**.

File: docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx [hunk 27]
```diff
@@ -0,0 +1,141 @@
...
+3. Choose the **Enable** operation and select **Progressive Rollout**.
+4. Configure the rollout and submit.
+
+:::caution Number of variations
+
+The progressive rollout only works for flags with **two variations**. 
+
+:::
+
+:::caution Disabled Flag behaviour
```
Comment (cre8ivejp) [inline docs/feature-flags/creating-feature-flags/auto-operation/rollout.mdx:43]:
```suggestion
:::info Disabled Flag behaviour
```

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**